docs/core-concepts/router.md

```diff
@@ -0,0 +1,146 @@
+# Router
+The router is responsible for matching routes to urls and updating the browser history. It also provides a way to navigate to routes, access the current route, and provides some routing utilities. See [Router](/api/types/Router.md) type for more information.
+
+## Creating a Router
+Create a router using the `createRouter` utility.
+
+```ts {8}
+import { createRoute, createRouter } from '@kitbag/router'
+
+const routes = [
+  createRoute(...),
+  createRoute(...),
+] as const
+
+const router = createRouter(routes)
+```
+
+## Installing
+Once you have created a router you can install it into a Vue application using the `use` method.
+
+```ts {8}
+import { createRouter } from '@kitbag/router'
+import { createApp } from 'vue'
+import App from './App.vue'
+
+const router = createRouter(...)
+const app = createApp(App)
+
+app.use(router)
+app.mount('#app')
+```
+
+## Type Safety
+Kitbag Router utilizes [declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html) to provide the internal types to match the actual router you're using.
+
+```ts
+declare module '@kitbag/router' {
+  interface Register {
+    router: typeof router
+  }
+}
+```
+
+## Router Options
+There are several options you can pass to the router to customize its behavior. See the [RouterOptions](/api/types/RouterOptions.md) type for more information and a list of options.
+
+```ts
+const router = createRouter(routes, {
+  base: '/app',
+})
+```
+
+## Router Instance
+The router created by `createRouter` can be used directly in your application. You can also access the router instance using the [useRouter](/composables/useRouter.md) composable within your components.
```

Comment:
as you might have picked up on by now, I'd prefer our core-concepts to be as concise as possible. I personally wouldn't even mention the fact that you _could_ import your router, let's just tell them about `useRouter` as a good pattern for accessing the installed router

Reply:
I agree on being concise but also complete. I don't want to have a core concept and a deep dive page about the same topic. 

Also agree that `useRouter` is the preferred way of getting the router instance. Which is why that's the example here. I didn't intent to imply you could or should export/import it. Just that the return value of `createRouter` IS the router instance.

Reply:
We're going to continue to work on this page. We both agree its needed but not great in its current state.